src/wp-includes/rest-api/endpoints/class-wp-rest-font-families-controller.php

```diff
@@ -0,0 +1,554 @@
+	 */
+	public function get_item_permissions_check( $request ) {
+		$post = $this->get_post( $request['id'] );
+		if ( is_wp_error( $post ) ) {
+			return $post;
+		}
+
+		if ( ! current_user_can( 'read_post', $post->ID ) ) {
+			return new WP_Error(
+				'rest_cannot_read',
+				__( 'Sorry, you are not allowed to access this font family.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Validates settings when creating or updating a font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string          $value   Encoded JSON string of font family settings.
+	 * @param WP_REST_Request $request Request object.
+	 * @return true|WP_Error True if the settings are valid, otherwise a WP_Error object.
+	 */
+	public function validate_font_family_settings( $value, $request ) {
+		$settings = json_decode( $value, true );
+
+		// Check settings string is valid JSON.
+		if ( null === $settings ) {
+			return new WP_Error(
+				'rest_invalid_param',
+				/* translators: %s: Parameter name: "font_family_settings". */
+				sprintf( __( '%s parameter must be a valid JSON string.' ), 'font_family_settings' ),
+				array( 'status' => 400 )
+			);
+		}
+
+		$schema   = $this->get_item_schema()['properties']['font_family_settings'];
+		$required = $schema['required'];
+
+		if ( isset( $request['id'] ) ) {
+			// Allow sending individual properties if we are updating an existing font family.
+			unset( $schema['required'] );
+
+			// But don't allow updating the slug, since it is used as a unique identifier.
+			if ( isset( $settings['slug'] ) ) {
+				return new WP_Error(
+					'rest_invalid_param',
+					/* translators: %s: Name of parameter being updated: font_family_settings[slug]". */
+					sprintf( __( '%s cannot be updated.' ), 'font_family_settings[slug]' ),
+					array( 'status' => 400 )
+				);
+			}
+		}
+
+		// Check that the font face settings match the theme.json schema.
+		$has_valid_settings = rest_validate_value_from_schema( $settings, $schema, 'font_family_settings' );
+
+		if ( is_wp_error( $has_valid_settings ) ) {
+			$has_valid_settings->add_data( array( 'status' => 400 ) );
+			return $has_valid_settings;
+		}
+
+		// Check that none of the required settings are empty values.
+		foreach ( $required as $key ) {
+			if ( isset( $settings[ $key ] ) && ! $settings[ $key ] ) {
+				return new WP_Error(
+					'rest_invalid_param',
+					/* translators: %s: Name of the empty font family setting parameter, e.g. "font_family_settings[slug]". */
+					sprintf( __( '%s cannot be empty.' ), "font_family_settings[ $key ]" ),
+					array( 'status' => 400 )
+				);
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Sanitizes the font family settings when creating or updating a font family.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string          $value   Encoded JSON string of font family settings.
+	 * @param WP_REST_Request $request Request object.
```

Comment:
There is no `$request` parameter here

```suggestion
	 * @param string $value Encoded JSON string of font family settings.
```

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/6034/commits/a0836f0f6b4fffc3259a2711f2ea43b02823a6e2